src/libertem/udf/base.py

```diff
@@ -62,6 +62,52 @@
 ExecutionPlan = Dict[ArrayBackend, Iterable["UDF"]]
 
 
+class ResultsForDataSet:
+    def __init__(
+        self,
+        gen: Generator[Tuple[Tuple["UDFData", ...], TaskProtocol], None, None],
+        params: "UDFParams",
+        params_handle,
+        executor: JobExecutor,
+        udfs: List["UDF"],
+    ):
+        self._gen = gen
+        self._params = params
+        self._params_handle = params_handle
+        self._executor = executor
+        self._udfs = udfs
+
+    def __iter__(self):
+        return self
+
+    def __next__(self) -> Tuple[Tuple["UDFData", ...], TaskProtocol]:
+        return next(self._gen)
+
+    def close(self):
+        self._gen.close()
+
+    def throw(self, exc: Exception):
+        return self._gen.throw(exc)
+
+    def update_parameters_experimental_full(self, parameters: List[Dict[str, Any]]):
+        log.debug("ResultsForDataSet.update_parameters_experimental_full: %s", parameters)
+        # update parameters on workers (via the executor):
+        self._params.update_parameters(parameters=parameters)
+        self._executor.scatter_update(self._params_handle, self._params)
+        # update UDFs on the main node:
+        for udf, params in zip(self._udfs, parameters):
+            udf._update_parameters(new_kwargs=params)
+
```

Comment:
This function is not documented and not covered by tests, if I see correctly? What is the status of it?